### PR TITLE
initVirtualMachineMap, Limit vmList request to avoid memory max limit

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
limit the memory consumption when kubemacpool initiates the vm mac map
by limiting the request to chunks of 100 vms.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RamLavi** [comment]
@qinqon should we backport to 4.8 stable branch? I'm leaning to yes

File: pkg/pool-manager/virtualmachine_pool.go [hunk 1]
```diff
@@ -270,61 +270,86 @@ func (p *PoolManager) initMacMapFromCluster(parentLogger logr.Logger) error {
...
-	err := result.Into(vms)
-	if err != nil {
-		return err
-	}
+// paginateVmsWithLimit performs a vm list request with pagination, to limit the amount of vms received at a time
+// and prevent taking too much memory.
+func (p *PoolManager) paginateVmsWithLimit(limit int64, vmsFunc func(pods *kubevirt.VirtualMachineList) error) error {
+	continueFlag := ""
+	for {
+		var result = p.kubeClient.ExtensionsV1beta1().RESTClient().Get().RequestURI(fmt.Sprintf("apis/kubevirt.io/v1/virtualmachines?limit=%v&continue=%v", limit, continueFlag)).Do(context.TODO())
```
Comment (qinqon) [inline pkg/pool-manager/virtualmachine_pool.go:278]: This feels a little fragile, we don't have the limit/continue option at kubevirt client right ?

**Review by qinqon — COMMENTED** [review]
Just one question

File: pkg/pool-manager/virtualmachine_pool.go (hunk 1 above)
Comment (RamLavi) [reply pkg/pool-manager/virtualmachine_pool.go:278]: This is the external api support [way](https://kubernetes.io/docs/reference/using-api/api-concepts/#retrieving-large-results-sets-in-chunks)..
I think once we'll incorporate the kubevirt client way then we'll be able to use the ListOptions ...

Comment (qinqon) [reply pkg/pool-manager/virtualmachine_pool.go:278]: Ack, can we add a `//FIXME` note about it ?

Comment (RamLavi) [reply pkg/pool-manager/virtualmachine_pool.go:279]: totally, [Done](https://github.com/k8snetworkplumbingwg/kubemacpool/compare/1064582b3be4f87a570056d764f1af7099fc7536..5bccedba8228960095a047dc893efbf87b2dc465)

**Pull request merged**